dumper.py

```diff
@@ -0,0 +1,141 @@
+import yaml
+import inspect
+from typing import *
+
+
+def parseargs(doc: str) -> List[Dict[str, str]]:
+    args = []
+    lines = [line.strip() for line in doc.splitlines()]
+    while lines:
+        line = lines.pop(0)
+        try:
+            name, desc = line.split(':', 1)
+        except ValueError:
+            pass
+        else:
+            name = name.strip()
+            desc = desc.strip()
+            type = None
+            if '(' in name:
+                name, rest = name.split('(', 1)
+                type, rest = rest.split(')', 1)
+                name = name.strip()
+            arg = {'name': name, 'desc': desc}
+            if type is not None:
+                arg['type'] = type
+            args.append(arg)
+    return args
+
+
+def parsereturns(doc: str) -> List[Dict[str, str]]:
+    rets = []
+    lines = [line.strip() for line in doc.splitlines()]
+    while lines:
+        line = lines.pop(0)
+        try:
+            type, desc = line.split(':', 1)
+        except ValueError:
+            pass
+        else:
+            type = type.strip()
+            desc = desc.strip()
+            ret = {'type': type, 'desc': desc}
+            rets.append(ret)
+    return rets
+
+
+def parse(doc: str) -> Dict[str, Any]:
+    res = {}
+    lines = [line.strip() for line in doc.splitlines()]
+    while lines:
+        line = lines.pop(0)
+        try:
+            key, value = line.split(':', 1)
+        except ValueError:
+            pass
+        else:
+            if key[:1].isupper():  # is a valid taichi doc key, e.g. Scope, Args, Returns
+                key = key.lower().strip()
+                if not value:  # multiline value
+                    values = []
+                    while lines:
+                        line = lines.pop(0)
+                        try:
+                            nextkey, nextvalue = line.split(':', 1)
+                        except ValueError:
+                            pass
+                        else:
+                            if nextkey[:1].isupper():  # is a valid taichi doc key
+                                lines.insert(0, line)
+                                break
+                        values.append(line)
+                    value = '\n'.join(values)  # we always enforce LF as linesep even on windows
```

Comment:
```suggestion
                    value = '\n'.join(values)
```
No need to emphasize this.